tests/rest/client/test_relations.py

```diff
@@ -651,6 +653,60 @@ def test_aggregation_get_event_for_thread(self):
             },
         )
 
+    def test_ignore_invalid_room(self):
+        """Test that we ignore invalid relations over federation."""
```

Comment:
> Why single out federation here, sorry? Because the CS API should reject this kind of event if a local client tries to send it?

Yes, exactly. We're somewhat assuming our server isn't generating "bad" events, but they could have before C-S had validation, but it is more likely to be from a buggy / malicious server over federation.